bottlecap/src/metrics/enhanced/lambda.rs

```diff
@@ -1,21 +1,29 @@
 use super::constants::{self, BASE_LAMBDA_INVOCATION_PRICE};
+use super::metric_data;
+use crate::proc::proc;
 use crate::telemetry::events::ReportMetrics;
 use dogstatsd::aggregator::Aggregator;
 use dogstatsd::metric;
 use dogstatsd::metric::{Metric, MetricValue};
+use std::collections::VecDeque;
 use std::env::consts::ARCH;
 use std::sync::{Arc, Mutex};
-use tracing::error;
+use tracing::{debug, error};
 
 pub struct Lambda {
     pub aggregator: Arc<Mutex<Aggregator>>,
     pub config: Arc<crate::config::Config>,
+    enhanced_metric_data: VecDeque<Vec<Box<dyn metric_data::EnhancedMetricData>>>,
```

Comment:
What's the decision on this type of architecture?

Reply:
Each invocation has a set of offsets/data that needs to be stored at the beginning of the invocation, which we would be storing in the Vec (currently it only holds NetworkData, but in the future would include others like CPUData, etc.)

The queue is there so that when the metrics are being sent after the PlatformReport event, which may be at the start of the subsequent invocation (at which point there would be two of each type of metric data), we only calculate the set of metrics for the invocation that has finished by popping off the Vec object at the front.

Reply:
Why are we using a mutable field for a trait? Can't the collection of data be a simple struct or vec?

Reply:
I just don't understand why we would all of those files for this use case. This look overly complex for what we actually have in our source code, this could be simplified even more

Reply:
I gave some advice here to Shreya, but I think the basic idea is, there will be many different implementations of these metrics, that will all have different behaviour, eg. different intervals they are sampled, different times they should be compared to previous values, and store different data. To me, that makes sense as a Trait with multiple different structs implementing it. That way, we don't need all the state/logic bundled together into a megastruct or enum. Now, I would  still suggest as a change here, which would be not to use a Dequeue and just have a regular vec/array, but to store all internal data/state for a metric type inside the struct itself, including if it needs to queue state.

Reply:
I had a chat with Shreya today, I showed the current `invocation/context.rs` in the Universal Instrumentation branch which contains a 5-context buffer which has relevant data for metrics. This way we can have some of this data living there, maybe we can have a broader chat with the team around how we handle this, I do agree we could have a trait which multiple structs could implement!